### PR TITLE
feat(RightSidebar): enforce tab navigation, show chat description in sidebar

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Ref #12800
* Follow-up of discussion in https://github.com/nextcloud/spreed/pull/12810#discussion_r1691396919

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
Group chat | --
image | image
image | image
Federated chat | --
image | image
image | image
1-1 chat | --
image | image
image | image

### 🚧 Tasks

- [x] Top bar title is cut off - keep it one line only

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/TopBar/TopBar.vue [hunk 1]
```diff
@@ -332,38 +312,31 @@ export default {
...
-	.description {
-		overflow: hidden;
-		text-overflow: ellipsis;
-		max-width: fit-content;
-		&__in-chat {
-			color: var(--color-text-maxcontrast);
-		}
+
+	&__status {
+		padding-inline: var(--default-grid-baseline);
```
Comment (ShGKme) [inline src/components/TopBar/TopBar.vue:337]: Maybe `2 * --default-grid-baseline`?

File: src/components/RightSidebar/RightSidebar.vue [hunk 2]
```diff
@@ -27,6 +28,12 @@
 			</NcCounterBubble>
 		</template>
 		<template #description>
+			<NcRichText v-if="conversation.description"
+				class="rich-text-description"
+				:text="conversation.description"
+				dir="auto"
+				autolink
+				use-extended-markdown />
```
Comment (ShGKme) [inline src/components/RightSidebar/RightSidebar.vue:36]: Nice!

File: src/components/RightSidebar/RightSidebar.vue [hunk 3]
```diff
@@ -12,6 +12,7 @@
 		:class="'active-tab-' + activeTab"
 		:toggle-classes="{ 'chat-button-sidebar-toggle': isInCall }"
 		:toggle-attrs="isInCall ? inCallToggleAttrs : undefined"
+		force-tabs
```
Comment (ShGKme) [inline src/components/RightSidebar/RightSidebar.vue:15]: Not sure about forced tabs... IMO, looked good without navigation when there was only on tab.

File: src/components/RightSidebar/RightSidebar.vue [hunk 4]
```diff
@@ -27,6 +28,12 @@
 			</NcCounterBubble>
 		</template>
 		<template #description>
+			<NcRichText v-if="conversation.description"
```
Comment (nickvergessen) [inline src/components/RightSidebar/RightSidebar.vue:31]: needs a max height somehow

**Comment by marcoambrosini** [comment]
@Antreesy what is the issue we're trying to solve here? At first glance I don't see improvements and only potential drawbacks

**Comment by Antreesy** [comment]
@marcoambrosini check the initial PR and discussion: https://github.com/nextcloud/spreed/pull/12810#discussion_r1691396919
Was suggested either decrease the font size in TopBar to fit two lines in 34px, or find another way, like move the description out of it

**Review by jancborchardt — CHANGES_REQUESTED** [review]
@Antreesy I agree with @marcoambrosini that this is mostly a potential drawback – the description was intentionally moved to the subline to be always visible, which is super helpful.

We can go for a third option: Increase the height of the header bar (aka not decrease it so much) to be able to fit 2 lines of text, to also be symmetric to any of the entries in the left navigation.

**Comment by ShGKme** [comment]
> We can go for a third option: Increase the height of the header bar (aka not decrease it so much) to be able to fit 2 lines of text, to also be symmetric to any of the entries in the left navigation.

@jancborchardt Then it won't be in line with the navigation toggle, navigation search, sidebar toggle, and sidebar title, which intentionally was forced to be the same everywhere in components.

**Comment by Antreesy** [comment]
See https://github.com/nextcloud/spreed/pull/12851
Looks okay for me with reduced padding and default line-height (at least text won't be trimmed)

**Comment by marcoambrosini** [comment]
I would downsize tue subline to 13px 
Screenshot 2024-07-30 at 13 13 57

We should create a new variable for it: `--font-size-small`

**Pull request closed without merging**